src/roles/actors.rs

```diff
@@ -376,13 +385,20 @@ decl_module! {
 
             let role_parameters = Self::ensure_role_parameters(actor.role)?;
 
-            Self::apply_unstake(actor.account.clone(), actor.role, actor.member_id, role_parameters.unbonding_period);
+            Self::apply_unstake(actor.account.clone(), actor.role, actor.member_id, role_parameters.unbonding_period, role_parameters.min_stake);
 
             Self::deposit_event(RawEvent::Unstaked(actor.account, actor.role));
         }
 
         pub fn set_role_parameters(role: Role, params: RoleParameters<T>) {
+            let new_stake = params.min_stake.clone();
             <Parameters<T>>::insert(role, params);
+            // Update locks for all actors in the role. The lock for each account is already until max_value
```

Comment:
Increasing the minimum staking amount while actors are already staked would presumably fail to update the lock for those who don't have excess balance. It's likely that you have to gracefully unstake these actors, or something?

Reply:
The locking mechanism should not fail. It is permitted to lock more than the current balance.

Reply:
That was new to me, thanks!